lib/ioc_container.dart

```diff
@@ -204,12 +207,14 @@ class IocContainerBuilder {
   void addSingletonAsync<T>(
```

Comment:
@jinyus you're on the right track here, but I think think this extension no longer makes sense

Others have complained that this extension is confusing, so I'm considering deleting this.

Reply:
I'm not sure I follow, how would you register a Database singleton that needs to be initialized asynchronously?

```dart
final builder = IocContainerBuilder()
    ..addSingletonAsync(
      (container) async => DatabaseService('connection_string').init(),
    )
```

Reply:
@jinyus this doesn't change at all. I have just deleted `addSingletonService`

You only need to delete changes to the `addSingletonService` because it no longer exists.

Reply:
ok, sounds good.